mathics/core/element.py

```diff
@@ -469,4 +469,17 @@ class BoxElement(ImmutableValueMixin, BaseElement):
     elements
     """
 
-    pass
+    def boxes_to_text(self, **options) -> str:
```

Comment:
It seems to me that instead of having three specific box formatters, there should be one  "box" routine that is passed the format which is either "text", "text", "mathml", or one an unlimited number of formatters.  

Yes, this would mean places were "boxes_to_x" is done, but I wonder how onerous that would be.

Reply:
Yes, the idea is to remove these `boxes_to_x` routines in favor of `mathics.format.[format].[method that takes a boxed expression and produces an string or bitstring of the appropriate format ]`

I kept this just for backward compatibility, and to make the PR a little bit smaller...

Reply:
The backwards compatibility I had forgotten about, so good catch. Yes, this is used for example in pymathics-graph and mathics-django.  

pymathics-graph that has to be revised, anyway. 

But still, there are two things that would be beneficial here:

* split out BoxElementMixin and ImmutableValueMixin into a separate module, say "mixin.py"
* Write a single box routine and have those specific box routines call that.

By doing this, you don't have to do an import inside each def. And we DRY the code.

For example:


```
from mathics.core.formatter import lookup_method

class ImmutableValueMixin:
    @property
    def is_literal(self) -> bool:
        """
        The value value can't change once it is set.
        """
        return True

class BoxElementMixin(ImmutableValueMixin):
    """
    The base class for all the boxed
    elements
    """

    def box(self, format: str, **options: dict) -> str:
        return lookup_method(self, format)(self, **options)

    
    def boxes_to_mathml(self, **options:dict) -> str:
        """For compatibility deprecated"""
        return self.box("mathml", **options)

    def boxes_to_tex(self, **options: dict) -> str:
        """For compatibility deprecated"""
        return self.box("tex", **options)

    def boxes_to_text(self, **options: dict) -> str:
        return self.box("text, **options)

```

Reply:
OK with the DRY part. However, I am not sure about the part about splitting mixings into a module. I checked it and even doing it,   it is not possible to avoid the circular import without doing it locally.  On the other hand, I do not understand the logic of having different mixins in one module from a conceptual point of view. Sometimes it could make sense to import `ImmutableValueMixin` (or `EvalMixn`) in contexts where `BoxElementMixin` is not involved.